src/app/assessment/assessment.component.ts

```diff
@@ -239,38 +279,52 @@ export class AssessmentComponent extends RouterEnter {
         }
       });
     }
-
     // save the submission/feedback
-    this.assessmentService.saveAnswers(assessment, answers, this.action)
-      .subscribe(result => {
+    console.log('saveAnswers', assessment, answers, this.action, this.submission.id);
+    this.assessmentService.saveAnswers(assessment, answers, this.action, this.submission.id ? this.submission.id : undefined).subscribe(
```

Comment:
Actually we can just pass `this.submission.id` here, it will be 0 if there's no submission.